### PR TITLE
Implements RenderInfoClass.

### PR DESCRIPTION
**Pull request by tomsons26**

### CODE REVIEW

File: src/w3d/renderer/rinfo.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#pragma once
+#include "always.h"
+#include <captainslog.h>
```
Comment (OmniBlade) [inline src/w3d/renderer/rinfo.h]: You don't appear to be logging anything from the header, so include this in the cpp file, not here.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: src/w3d/renderer/rinfo.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+void RenderInfoClass::Push_Material_Pass(MaterialPassClass *matpass)
+{
+    if (m_additionalMaterialPassCount <= ADDITIONAL_MATERIAL_PASSES) {
+        if (matpass != nullptr) {
+            // TODO temp until MaterialPassClass is in
+            ((RefCountClass *)matpass)->Add_Ref();
+        }
+        m_additionalMaterialPassArray[m_additionalMaterialPassCount++] = matpass;
+    } else {
+        ++field_A8;
```
Comment (OmniBlade) [inline src/w3d/renderer/rinfo.cpp:58]: Is this used anywhere else? It looks to me like is a counter for additional passes that are requested, but there is no room for, perhaps a debug counter for logging if additional passes are being requested?

Comment (tomsons26) [reply src/w3d/renderer/rinfo.cpp:58]: Haven't been able to find

File: src/w3d/renderer/rinfo.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    int Additional_Pass_Count() { return m_additionalMaterialPassCount; }
+    MaterialPassClass *Peek_Additional_Pass(int i) const;
+
+    void Push_Override_Flags(RINFO_OVERRIDE_FLAGS flags);
+    void Pop_Override_Flags();
+    RINFO_OVERRIDE_FLAGS &Current_Override_Flags();
+
+public:
+    CameraClass &m_camera;
+    int field_4; // float - fog_scale, fog_start, fog_end or according to BFME2 Vector3 FogColor
```
Comment (OmniBlade) [inline src/w3d/renderer/rinfo.h]: Can we make these float if we know they are float?

File: src/w3d/renderer/rinfo.cpp [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    for (int i = 0; i < ADDITIONAL_MATERIAL_PASSES; ++i) {
+        m_additionalMaterialPassArray[i] = nullptr;
+    }
+
+    for (int i = 0; i < MAX_OVERRIDE_FLAG_LEVEL; ++i) {
+        m_overrideFlag[i] = RINFO_OVERRIDE_DEFAULT;
+    }
+}
+
+inline RenderInfoClass::~RenderInfoClass()
```
Comment (OmniBlade) [inline src/w3d/renderer/rinfo.cpp]: No point marking things inline in the cpp file.

**Review by OmniBlade — CHANGES_REQUESTED** [review]

File: src/w3d/renderer/rinfo.h (hunk 3 above)
Comment (tomsons26) [reply src/w3d/renderer/rinfo.h]: we don't know anything, got no info on it, its just speculation based on known things
whenever/if ever we encounter it we figure out what to do

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**